src/golang/lib/workflow/artifact/artifact.go

```diff
@@ -376,31 +378,44 @@ func (a *ArtifactImpl) SampleContent(ctx context.Context) ([]byte, bool, error)
 
 	// For table types, we returns a down-sampled table when possible
 	if metadata.SerializationType == shared.TableSerialization {
-		// The over-simplified type for table orient.
+		readBuffer := buffer.NewBufferFileFromBytes(content)
+		parquetReader, err := reader.NewParquetReader(
+			readBuffer,
+			nil,
+			4, //parallel number.
+		)
+		if err != nil {
+			return nil, false, err
+		}
+		parquetTable, err := parquetReader.ReadByNumber(sampleTableRow)
+		if err != nil {
+			return nil, false, err
+		}
 		type table struct {
 			Schema map[string]interface{} `json:"schema"`
 			Data   []interface{}          `json:"data"`
 		}
-
 		var t table
+		schema := make(map[string]interface{})
 
-		err := json.Unmarshal(content, &t)
-		if err != nil {
-			return nil, false, err
-		}
+		schemaFromParquet, _ := parquetReader.SchemaHandler.GetType(parquetReader.SchemaHandler.GetRootInName())
 
-		if len(t.Data) <= sampleTableRow {
-			// If the table is small, return the original content
-			return content, false, nil
+		for i := 0; i < schemaFromParquet.NumField(); i++ {
+			schema[schemaFromParquet.Field(i).Name] = schemaFromParquet.Field(i).Type.String()[1:]
 		}
 
-		t.Data = t.Data[:sampleTableRow]
-		downsampledContent, err := json.Marshal(t)
+		t.Data = parquetTable
+		t.Schema = schema
+
+		jsonTable, err := json.Marshal(t)
```

Comment:
Instead of JSON marshaling the Parquet data, can we convert the Parquet data into JSON. This would allow us to avoid any SDK or UI side changes for reading table artifacts for now.